File: T1.SqlSharp/T1.SqlSharp/Expressions/SqlConstraintColumn.cs
namespace T1.SqlSharp.Expressions;

public class SqlConstraintColumn : ISqlExpression 
{
    public SqlType SqlType { get; } = SqlType.Constraint;
    public TextSpan Span { get; set; } = new();
    public void Accept(SqlVisitor visitor)
    {
        visitor.Visit_ConstraintColumn(this);
    }

    public string ColumnName { get; set; } = string.Empty;
    public string Order { get; set; } = string.Empty;

    public string ToSql()
    {
        return $"{ColumnName} {Order}";
    }
}

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlConstraintColumn.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlExpression, SqlVisitor } from './ISqlExpression';

export class SqlConstraintColumn implements ISqlExpression {
    SqlType: SqlType = SqlType.Constraint;
    Span: TextSpan = new TextSpan();
    ColumnName: string = '';
    Order: string = '';

    Accept(visitor: SqlVisitor): void {
        visitor.Visit_ConstraintColumn(this);
    }

    ToSql(): string {
        return `${this.ColumnName} ${this.Order}`;
    }
} 